src/libpgmoneta/wf_hot_standby.c

```diff
@@ -138,195 +127,241 @@ hot_standby_execute(char* name __attribute__((unused)), struct art* nodes)
+               {
+                  f = pgmoneta_append_char(f, '/');
+               }
+               f = pgmoneta_append(f, deleted_iter->key);
+
+               if (pgmoneta_exists(f))
+               {
+                  pgmoneta_delete_file(f, workers);
+               }
+               else
+               {
+                  pgmoneta_log_debug("%s doesn't exists", f);
+               }
+
+               free(f);
+               f = NULL;
             }
-            from = pgmoneta_append(from, "data/");
-            from = pgmoneta_append(from, changed_iter->key);
 
-            to = pgmoneta_append(to, destination);
-            if (!pgmoneta_ends_with(to, "/"))
+            while (pgmoneta_art_iterator_next(changed_iter))
             {
-               to = pgmoneta_append_char(to, '/');
+               from = pgmoneta_append(from, source);
+               if (!pgmoneta_ends_with(from, "/"))
+               {
+                  from = pgmoneta_append_char(from, '/');
+               }
+               from = pgmoneta_append(from, "data/");
+               from = pgmoneta_append(from, changed_iter->key);
+
+               to = pgmoneta_append(to, destination);
+               if (!pgmoneta_ends_with(to, "/"))
+               {
+                  to = pgmoneta_append_char(to, '/');
+               }
+               to = pgmoneta_append(to, changed_iter->key);
+
+               pgmoneta_log_trace("hot_standby changed: %s -> %s", from, to);
+
+               pgmoneta_copy_file(from, to, workers);
+
+               free(from);
+               from = NULL;
+
+               free(to);
+               to = NULL;
             }
-            to = pgmoneta_append(to, changed_iter->key);
-
-            pgmoneta_log_trace("hot_standby changed: %s -> %s", from, to);
-
-            pgmoneta_copy_file(from, to, workers);
 
-            free(from);
-            from = NULL;
-
-            free(to);
-            to = NULL;
+            while (pgmoneta_art_iterator_next(added_iter))
+            {
+               from = pgmoneta_append(from, source);
+               if (!pgmoneta_ends_with(from, "/"))
+               {
+                  from = pgmoneta_append_char(from, '/');
+               }
+               from = pgmoneta_append(from, "data/");
+               from = pgmoneta_append(from, added_iter->key);
+
+               to = pgmoneta_append(to, destination);
+               if (!pgmoneta_ends_with(to, "/"))
+               {
+                  to = pgmoneta_append_char(to, '/');
+               }
+               to = pgmoneta_append(to, added_iter->key);
+
+               pgmoneta_log_trace("hot_standby new: %s -> %s", from, to);
+
+               pgmoneta_copy_file(from, to, workers);
+
+               free(from);
+               from = NULL;
+
+               free(to);
+               to = NULL;
+            }
          }
-
-         while (pgmoneta_art_iterator_next(added_iter))
+         else // incremental or  number of backups < 2
```

Comment:
Use `/* */` as comment inside the block, not on the block

Reply:
rewrited it :D